dash/development/base_component.py

```diff
@@ -280,6 +287,55 @@ def traverse_with_paths(self):
                     for p, t in i.traverse_with_paths():
                         yield "\n".join([list_path, p]), t
 
+    def validate(self):
+        # Make sure arguments have valid values
+        DashValidator.set_component_class(Component)
+        validator = DashValidator(
+            self._schema,
+            allow_unknown=True,
+        )
+        args = {
+            k: self.__dict__[k]
+            for k in self.__dict__['_prop_names']
+            if k in self.__dict__.keys()
+        }
+        valid = validator.validate(args)
+        if not valid:
+            # pylint: disable=protected-access
+            error_message = dedent("""\
+
+                A Dash Component was initialized with invalid properties!
+
+                Dash tried to create a `{component_name}` component with the
+                following arguments, which caused a validation failure:
+
+                ***************************************************************
+                {component_args}
+                ***************************************************************
+
+                The expected schema for the `{component_name}` component is:
+
+                ***************************************************************
+                {component_schema}
+                ***************************************************************
+
+                The errors in validation are as follows:
+
+
+            """).format(
```

Comment:
For those big templates string, I like to put them as global as constant at the top of the file instead of polluting the function. Then the format can be on a single line. You also don't need to dedent when it's at the top level.

Reply:
Sounds good